bigbang/ingress/w3c.py

```diff
@@ -0,0 +1,614 @@
+import email
+import logging
+import os
+import re
+import subprocess
+import time
+import warnings
+import mailbox
+from mailbox import mboxMessage
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple, Union
+from urllib.parse import urljoin, urlparse
+
+import numpy as np
+import pandas as pd
+import requests
+import yaml
+from bs4 import BeautifulSoup
+from tqdm import tqdm
+
+from config.config import CONFIG
+
+from bigbang.bigbang_io import MessageIO, ListIO, ArchiveIO
+from bigbang.ingress import (
+    AbstractArchive,
+    AbstractList,
+    AbstractMessageParser,
+)
+from bigbang.utils import (
+    get_website_content,
+    set_website_preference_for_header,
+    get_auth_session,
+    get_paths_to_files_in_directory,
+    get_paths_to_dirs_in_directory,
+)
+
+filepath_auth = CONFIG.config_path + "authentication.yaml"
+directory_project = str(Path(os.path.abspath(__file__)).parent.parent)
+logging.basicConfig(
+    filename=directory_project + "/w3c.scraping.log",
+    filemode="w",
+    level=logging.INFO,
+    format="%(asctime)s %(message)s",
+)
+logger = logging.getLogger(__name__)
+
+
+class W3CMessageParserWarning(BaseException):
+    """Base class for Archive class specific exceptions"""
+
+    pass
+
+
+class W3CListWarning(BaseException):
+    """Base class for Archive class specific exceptions"""
+
+    pass
+
+
+class W3CArchiveWarning(BaseException):
+    """Base class for Archive class specific exceptions"""
+
+    pass
+
+
+class W3CMessageParser(AbstractMessageParser, email.parser.Parser):
+    """
+    This class handles the creation of an mailbox.mboxMessage object
+    (using the from_*() methods) and its storage in various other file formats
+    (using the to_*() methods) that can be saved on the local memory.
+
+    Parameters
+    ----------
+    website : Set 'True' if messages are going to be scraped from websites,
+        otherwise 'False' if read from local memory. This distinction needs to
+        be made if missing messages should be added.
+    url_login : URL to the 'Log In' page.
+    url_pref : URL to the 'Preferences'/settings page.
```

Comment:
I am confused why a Parser object needs login information.
Is this responsible both for accessing the data on-line, and then also parsing it into a a standard format?

Maybe these functionalities should be separated or the class renamed to make it clear what the class's full scope is.

Reply:
For the 3GPP mailing archive one needs to create an account to access all the messages, otherwise all reply-to header fields just show `<[login to unmask]>`. But for W3C this argument is not necessary thus I should remove them from the `/ingress/w3c.py` file.

Reply:
removed those unnecessary method arguments for W3C in this PR